pkg/distributor/distributor.go

```diff
@@ -1119,7 +1119,14 @@ func (d *Distributor) forwardSamples(ctx context.Context, userID string, ts []mi
 func (d *Distributor) Push(ctx context.Context, req *mimirpb.WriteRequest) (*mimirpb.WriteResponse, error) {
 	pushReq := push.NewParsedRequest(req)
 	pushReq.AddCleanup(func() { mimirpb.ReuseSlice(req.Timeseries) })
-	return d.PushWithMiddlewares(ctx, pushReq)
+
+	return d.GetPushFunc(nil)(ctx, pushReq)
+}
+
+// GetPushFunc returns push.Func that can be used by push handler.
+// Wrapper, if not nil, is added to the list of distributor middlewares.
+func (d *Distributor) GetPushFunc(distributorWrapper func(next push.Func) push.Func) push.Func {
+	return d.wrapPushWithMiddlewares(distributorWrapper, d.push)
```

Comment:
Another non-blocking nit:

Am I understanding it correctly that we only need `GetPushFunc()` to be able to mock `d.push` in tests? 
Usually when creating these wrappers that are only necessary for mocking I'd give them the same name as the function that they wrap, with the only difference being that they're exported while the wrapped function remains unexpected. 

That way it's clear that they're basically the same with the only difference being that one is meant to be used from inside the package and the other from outside.

Reply:
Feel free to leave it as it is if you disagree with me.

Reply:
I personally find it confusing to have two methods with the same name (one exported, one not), but different signature. I thought `GetPushFunc` would better describe what we're doing here. The fact that we're wrapping with middlewares is internal detail of distributor, not something we need to be public about.